modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -1130,6 +1184,10 @@ std::ostream& operator<<(std::ostream&, const Quat<S>&);
 
 using Quatd = Quat<double>;
 using Quatf = Quat<float>;
+using std::cos;
```

Comment:
We have added the `std::` before these trigonometric functions during implementation, but some other files that also in `cv` namespace are not, like affine.hpp and warpers_inl.hpp. They just use it directly.

So we finally add `using std::*` to solve the error(see below) raising from compilation with the entire OpenCV.

error is something like this

```
  /home/kong/opencv/modules/core/include/opencv2/core/affine.hpp:502:26: error: no matching function for call to ‘acos(double&)’
    double theta = acos(c);
     ...
 /home/kong/opencv/modules/core/include/opencv2/core/quaternion.inl.hpp:494:16: note:   template argument deduction/substitution failed:
/home/kong/opencv/modules/stitching/include/opencv2/stitching/detail/warpers_inl.hpp:359:27: note:   mismatched types ‘const cv::Quat<S>’ and ‘float’
     v = scale * r * sin(u_);
```
P.S. This error will not occur without adding enum
@savuor @alalek

Reply:
Use
```
-double theta = acos(c);
+double theta = std::acos(c);
```

`std::` directly instead.

Avoid using of "using" in public headers.

Reply:
This line of code is in other file（affine.hpp）. Can I directly change it?

Reply:
I've changed it in affine.hpp and warpers_inl.hpp, please have a look. Thanks